src/app/src/pages/eval/components/ResultsFilters/FiltersForm.tsx

```diff
@@ -189,15 +202,15 @@ function Filter({
    */
   const handleLogicOperatorChange = useCallback(
     (filterLogicOperator: ResultsFilter['logicOperator']) => {
-      if (index === 1) {
-        // If this is the second filter, update all filters to have the same logic operator
+      if (value.sortIndex === 1) {
+        // If this is the second filter (by sortIndex), update all filters to have the same logic operator
         updateAllFilterLogicOperators(filterLogicOperator);
       } else {
         // Otherwise just update this filter
         updateFilter({ ...value, logicOperator: filterLogicOperator });
       }
```

Comment:
is this case ever reached?

Reply:
It shouldn't be because the selector is disabled when the sort index > 1.